MonstieWash/Assets/Scripts/MonsterAi/MonsterBrain.cs

```diff
@@ -0,0 +1,308 @@
+            }
+        }
+    }
+
+    /// <summary>
+    /// Calls other function maintain limit for each mood currently active.
+    /// </summary>
+    private void maintainLimits()
+    {
+        for (int i = 0; i < activeMoods.Count; i++)
+        {
+            maintainLimit(i);
+        }
+    }
+
+    /// <summary>
+    /// Checks if mood given by its current index in active list has gone over or under its limits and then corrects it back.
+    /// </summary>
+    /// <param name="moodInt"> Given index of desired mood in active moods.</param> 
+    private void maintainLimit(int moodInt)
+    {
+        //Check upper limit and set back if over.
+        if (activeMoods[moodInt] > moodData[moodInt].MoodUpperLimit) activeMoods[moodInt] = moodData[moodInt].MoodUpperLimit;
+        //Check lower limit and set back if under.
+        if (activeMoods[moodInt] < moodData[moodInt].MoodLowerLimit) activeMoods[moodInt] = moodData[moodInt].MoodLowerLimit;
+    }
+
+    /// <summary>
+    /// Takes all moodtypes and stores them in activeMood dictionary, where first value is their index in moodtypes list and the second is their current value;
+    /// </summary>
+    private void loadMoods()
+    {
+        for(int i = 0; i < moodData.Count; i++)
+        {
+            activeMoods.Add(i, moodData[i].MoodStartingPoint);
+            activeMoodNames.Add(moodData[i].MoodName,i);
+        }
+    }
+
+
+    /// <summary>
+    /// Updates ui textbox with information useful to debugging.
+    /// </summary>
+    private void updateDebugText()
+    {
+        DebugUi.text = "";
+        for (int i = 0; i < activeMoods.Count; i++)
+        {
+            DebugUi.text += "MoodName: " + moodData[i].MoodName + " MoodValue: " + Mathf.FloorToInt(activeMoods[i]).ToString() + "\n" + 
+            " Mood Lower/Upper Limits: " + moodData[i].MoodLowerLimit.ToString() + "/" + moodData[i].MoodUpperLimit.ToString();
+            DebugUi.text += "\n";
+            DebugUi.text += "\n";
+        }
+
+    }
+
+
+    /// <summary>
+    /// Returns the index of a moodtype within active moods by its string name.
+    /// </summary>
+    /// <param name="name"> The name of the desired mood index</param>
+    /// <returns></returns>
+    /// <exception cref="System.Exception"> When no mood exists with that name </exception>
+    private int accessActiveMoodIndexByName(string name)
+    {
+        return activeMoodNames[name];
+    }
+
+    /// <summary>
+    /// Returns the index of a moodtype within active moods by a scriptable object reference.
+    /// </summary>
+    /// <param name="name"> The name of the desired mood index</param>
+    /// <returns></returns>
+    /// <exception cref="System.Exception"> When no mood exists with that name </exception>
+    private int accessActiveMoodIndexByMoodType(MoodType refMT)
+    {
+        return activeMoodNames[refMT.MoodName];
+    }
```

Comment:
These can be modified to use overloading to reduce the number of functions in the class.
```suggestion
    /// <summary>
    /// Returns the index of a moodtype within active moods by its string name.
    /// </summary>
    /// <param name="name"> The name of the desired mood index</param>
    /// <returns></returns>
    /// <exception cref="System.Exception"> When no mood exists with that name </exception>
    private int accessActiveMoodIndex(string name)
    {
        return activeMoodNames[name];
    }

    /// <summary>
    /// Returns the index of a moodtype within active moods by a scriptable object reference.
    /// </summary>
    /// <param name="name"> The name of the desired mood index</param>
    /// <returns></returns>
    /// <exception cref="System.Exception"> When no mood exists with that name </exception>
    private int accessActiveMoodIndex(MoodType refMT)
    {
        return activeMoodNames[refMT.MoodName];
    }
```